### PR TITLE
Flexible SaveWrapper

### PR DESCRIPTION
**Pull request by JamesLefrere**
- Make SaveWrapper stateless (that is, aside from being `Ownable`) so it can be flexible about assets supported
- Define pattern for discrete Hardhat tasks with params
- Disable some eslint rules

Tested on Ropsten, including a test sBTC/WETH pair on Uniswap.

### CODE REVIEW

File: tasks/deployAaveIntegration.ts [hunk 1]
```diff
@@ -30,6 +30,8 @@ task("deployAaveIntegration", "Deploys an instance of AaveV2Integration contract
         addresses.nexus,
         addresses.mAsset,
         addresses.aave,
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
```
Comment (JamesLefrere) [inline tasks/deployAaveIntegration.ts:34]: I guess the ABI has changed here

File: contracts/savings/peripheral/SaveWrapper.sol [hunk 2]
```diff
@@ -18,125 +19,205 @@ interface IBoostedSavingsVault {
...
+     * @param _amountOutMin   Min uniswap output in bAsset units
+     * @param _path           Sell path on Uniswap (e.g. [WETH, DAI])
+     * @param _minOutMStable  Min amount of mAsset to receive
+     * @param _stake          Add the imAsset to the Savings Vault?
      */
     function saveViaUniswapETH(
+        address _mAsset,
+        address _save,
+        address _vault,
+        address _uniswap,
```
Comment (JamesLefrere) [inline contracts/savings/peripheral/SaveWrapper.sol:99]: Theoretically this could work with SushiSwap, I think.

**Review by alsco77 — APPROVED** [review]
Nice work

File: contracts/savings/peripheral/SaveWrapper.sol (hunk 2 above)
Comment (alsco77) [reply contracts/savings/peripheral/SaveWrapper.sol:99]: Yep

**Pull request merged**